test/test_data/test_validation.py

```diff
@@ -530,3 +534,77 @@ def test_big_dataset_encoding(self):
         np.testing.assert_array_equal(x['carbon'].to_numpy(), x_t[:, 3])
 
         return
+
+    def test_join_and_check(self):
+        validator = InputValidator()
+
+        # Numpy Testing
+        y = np.array([2, 2, 3, 4, 5])
+        y_test = np.array([3, 4, 5, 6, 1])
+
+        joined = validator.join_and_check(y, y_test)
+        np.testing.assert_array_equal(
+            joined,
+            np.array([2, 2, 3, 4, 5, 3, 4, 5, 6, 1])
+        )
+
+        validator.validate_target(joined, is_classification=True)
+        y_encoded = validator.validate_target(y)
+        y_test_encoded = validator.validate_target(y_test)
+
+        # If a common encoding happened, then common elements
+        # should have a common encoding
+        self.assertEqual(y_encoded[2], y_test_encoded[0])
+
+        # Pandas Testing
+        validator = InputValidator()
+        joined = validator.join_and_check(
+            pd.DataFrame(y),
+            pd.DataFrame(y_test)
+        )
+        np.testing.assert_array_equal(
+            joined,
+            pd.DataFrame([2, 2, 3, 4, 5, 3, 4, 5, 6, 1])
+        )
+
+        # List Testing
+        validator = InputValidator()
+        joined = validator.join_and_check(
+            [2, 2, 3, 4, 5],
+            [3, 4, 5, 6, 1]
+        )
+        np.testing.assert_array_equal(
+            joined,
+            [2, 2, 3, 4, 5, 3, 4, 5, 6, 1]
+        )
+
+    def test_big_dataset_encoding2(self):
+        """
+        Makes sure that when there are multiple classes,
+        and test/train targets differ, we proactively encode together
+        the data between test and train
+        """
+        X, y = sklearn.datasets.fetch_openml(data_id=183, return_X_y=True, as_frame=True)
+        X_train, X_test, y_train, y_test = sklearn.model_selection.train_test_split(
+            X,
+            y,
+            random_state=1
+        )
+
+        # Make sure this test makes sense, so that y_test
+        # and y_train have different classes
+        all_classes = set(np.unique(y_test)).union(set(np.unique(y_train)))
+        elements_in_test_only = np.setdiff1d(np.unique(y_test), np.unique(y_train))
+        self.assertTrue(elements_in_test_only)
```

Comment:
I'm not sure if this works: https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertTrue

The element is first cast to bool and then checked whether it is True. How about `self.assertGreater(len(elements_in_test_only), 0)`?

Reply:
understood, I will change it in the next push